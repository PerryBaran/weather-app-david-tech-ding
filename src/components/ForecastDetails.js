import React from "react";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details_date">{formattedDate}</div>
      <div className="forecast-details_max-temp">
        <strong>Max temperature:</strong> {temperature.max}&deg;C
      </div>
      <div className="forecast-details_min-temp">
        <strong>Min temperature:</strong> {temperature.min}&deg;C
      </div>
      <div className="forecast-details_humidity">
        <strong>Humidity:</strong> {humidity} %
      </div>
      <div className="forecast-details_wind-speed">
        <strong>Wind Speed:</strong> {wind.speed} mph
      </div>
      <div className="forecast-details_wind-direction">
        <strong>Wind Speed:</strong> {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;
