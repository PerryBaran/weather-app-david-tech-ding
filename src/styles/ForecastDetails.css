.forecast-details{
    text-align: center;
    margin: 2em;
}

.forecast-details > div {
    padding-bottom: 0.5em;
}