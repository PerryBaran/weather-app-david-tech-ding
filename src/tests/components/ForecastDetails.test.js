import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: 20,
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 10,
      direction: "n",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details_date");
    expect(getByText("12°C")).toHaveClass("forecast-details_min-temp");
    expect(getByText("22°C")).toHaveClass("forecast-details_max-temp");
    expect(getByText("20 %")).toHaveClass("forecast-details_humidity");
    expect(getByText("10 mph")).toHaveClass("forecast-details_wind-speed");
    expect(getByText("n")).toHaveClass("forecast-details_wind-direction");
  });
});
