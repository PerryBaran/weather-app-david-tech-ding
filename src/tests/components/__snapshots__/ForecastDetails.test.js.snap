// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details_date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details_max-temp"
    >
      <strong>
        Max temperature:
      </strong>
       22°C
    </div>
    <div
      class="forecast-details_min-temp"
    >
      <strong>
        Min temperature:
      </strong>
       12°C
    </div>
    <div
      class="forecast-details_humidity"
    >
      <strong>
        Humidity:
      </strong>
       20 %
    </div>
    <div
      class="forecast-details_wind-speed"
    >
      <strong>
        Wind Speed:
      </strong>
       10 mph
    </div>
    <div
      class="forecast-details_wind-direction"
    >
      <strong>
        Wind Speed:
      </strong>
       n
    </div>
  </div>
</DocumentFragment>
`;
